tests/test_type_inference/test_literals.py

```diff
@@ -33,6 +33,10 @@ def test_homogeneous_lists(lst):
 def test_random_lists(lst):
     """Test List nodes representing a list of values of different primitive types."""
     assume(not isinstance(lst[0], type(lst[1])))
+    # bools and ints become unified; assume we have one or the other for this test case.
```

Comment:
This check is incorrect. Fix the `assume` above so that `lst[1]` is not an instance of `type(lst[0])` either.

Same comment below. Don't use `==` to compare types.